### DISCUSSION TITLE
How to reduce stream delay from webcam

### DISCUSSION THREAD

**Discussion by Saifer707** [bluenviron/mediamtx#1421] [Questions] [answered]
Hi everyone,

i'm trying to use this amazing rtsp server and i am able to stream my webcam on demand with this command:

`ffmpeg -f v4l2 -i /dev/video0 -pix_fmt yuv420p -preset ultrafast -b:v 600k -f rtsp rtsp://localhost:$RTSP_PORT/$RTSP_PATH`

However, i get a delay between what webcam captures and what stream shows. Is there a way to reduce or better remove this lag? 
I've already tried with -rtsp_transport option and udp value, but it seems not work. 

I need to use rtsp protocol only, i can't change that for my own purposes.

Thank you all!

**Comment by aler9** [comment, marked as answer]
Hello, try adding `-tune zerolatency`:

```
ffmpeg -f v4l2 -i /dev/video0 -pix_fmt yuv420p -preset ultrafast -tune zerolatency -b:v 600k -f rtsp rtsp://localhost:$RTSP_PORT/$RTSP_PATH
```

Furthermore, if you're using VLC to read the stream, there's a setting in the `Open network stream` dialog, check `Show more options` and edit the `Caching` value - decrease caching down in order to decrease latency.

**Reply by Saifer707** [reply]
Thank you so much! 

Set -tune only doesn't work using VLC for reading, but decreasing also cache value from VLC produces 0 latency! 

Thank you!

[a comment by an automated account is not shown]
